src/plugins/intel_cpu/src/nodes/memory.cpp

```diff
@@ -462,12 +462,16 @@ MemoryNodeVirtualEdge::Holder* MemoryNodeVirtualEdge::registerOutput(MemoryOutpu
     return &holder;
 }
 
-void MemoryNodeVirtualEdge::remove(MemoryNode * node, Holder* holder) {
+void MemoryNodeVirtualEdge::remove(MemoryNode* node, Holder* holder) {
     std::lock_guard<std::mutex> lock{MemoryNodeVirtualEdge::holderMutex};
     if (nullptr != holder) {
-        InferenceEngine::details::erase_if(*holder, [&](const Holder::value_type & it){
-            return it.second == node;
-        });
+        for (auto it = std::begin(*holder); it != std::end(*holder);) {
```

Comment:
Lets use std library. Smt like:
```
holder.erase(std::remove_if(holder.begin(), holder.end(), 
                                   [&](const Holder::value_type & it){
                                       return it.second == node;
                                   }),
                    holder.end());
```

Reply:
I think is not possible with maps. In C++20 there is `std::erase_if` for maps to solve it.

Reply:
agree, it will be better to migrate ie::details::erase_if to ov::util::erase_if, which will be easier to update and also can be used by other components if needed.